torch/utils/data/dataloader.py

```diff
@@ -792,6 +888,193 @@ def _next_data(self):
         return data
 
 
+class _StatefulBaseDataLoaderIter(_BaseDataLoaderIter):
+    """Base class for stateful dataloader iterators."""
+
+    def __init__(self, loader: DataLoader) -> None:
```

Comment:
The stateful iterators should take `next_iter_state`

```suggestion
    def __init__(self, loader: DataLoader, next_iter_state: Optional[dict[str, Any]] = None) -> None:
```

Reply:
Thanks, fixed.